src/actions/albums.js

```diff
@@ -45,6 +48,30 @@ export const fetchAlbums = (mangoIndex) => {
   }
 }
 
+// Returns albums photos informations from the album ID
+export const fetchAlbumPhotosStatsById = (albumId) => {
+  return async (dispatch) => {
+    dispatch({type: FETCH_CURRENT_ALBUM_PHOTOS})
+    try {
+      let album = await cozy.client.data.find(ALBUM_DOCTYPE, albumId)
+      const photosIds = await cozy.client.data.listReferencedFiles(album)
+      let fetchedPhotos = []
+      for (let index in photosIds) {
+        const photo = await cozy.client.files.statById(photosIds[index])
+        fetchedPhotos.push(photo)
+      }
+      album.photos = fetchedPhotos
+      dispatch({type: FETCH_CURRENT_ALBUM_PHOTOS_SUCCESS, album})
+      return fetchedPhotos
+    } catch (fetchError) {
+      throw new Error(fetchError.response
```

Comment:
These 3 lines of error handling are repeated several times in this file. Maybe a small refactoring here?